msal4j-sdk/src/main/java/com/microsoft/aad/msal4j/ConfidentialClientApplication.java

```diff
@@ -119,32 +120,25 @@ protected ClientAuthentication clientAuthentication() {
             final Date currentDateTime = new Date(System.currentTimeMillis());
             final Date expirationTime = ((PrivateKeyJWT) clientAuthentication).getJWTAuthenticationClaimsSet().getExpirationTime();
             if (expirationTime.before(currentDateTime)) {
-                //The asserted private jwt with the client certificate can expire so rebuild it when the
-                clientAuthentication = buildValidClientCertificateAuthority();
+                if (Authority.detectAuthorityType(this.authenticationAuthority.canonicalAuthorityUrl()) == AuthorityType.ADFS) {
+                    clientAuthentication = buildValidClientCertificateAuthority(true);
+                } else  {
+                    clientAuthentication = buildValidClientCertificateAuthority(false);
+                }
             }
         }
         return clientAuthentication;
     }
 
-    private ClientAuthentication buildValidClientCertificateAuthority() {
-        ClientAssertion clientAssertion = JwtHelper.buildJwt(
-                clientId(),
-                clientCertificate,
-                this.authenticationAuthority.selfSignedJwtAudience(),
-                sendX5c,
-                false);
-        return createClientAuthFromClientAssertion(clientAssertion);
-    }
-
     //The library originally used SHA-1 for thumbprints as other algorithms were not supported server-side,
     //  and while support for SHA-256 has been added certain flows still only allow SHA-1
```

Comment:
super duper nit: add a , after added. In my mind, it makes sense to indicate a pause to break up the sentence - please feel free to ignore this if you want

Reply:
In the latest commit I fixed the grammar and adjusted the wording a bit.